### ISSUE TITLE
Problems with NCSS and WMS in v5

### ISSUE THREAD

**Issue by marceloandrioni** [Unidata/tds#169]
Hello, I was doing some tests with the newly released  v5 and I think i found some problems with the NCSS <s>and WMS (embedded ncWMS)</s>.

Environment:
OS: Linux 5.4.0-88-generic x86_64 x86_64 x86_64 GNU/Linux
JDK: jdk-11.0.12_linux-x64
Tomcat: apache-tomcat-8.5.72
TDS: v5.0
 
How to reproduce the problem:

I installed the environment following the [docs](https://docs.unidata.ucar.edu/tds/5.0/userguide/index.html) and added a simple file ([cmems_forecast_20210101.zip](https://github.com/Unidata/tds/files/7355492/cmems_forecast_20210101.zip)) with 3 time-steps and 3 vertical levels (from [CMEMS](https://resources.marine.copernicus.eu/product-detail/GLOBAL_ANALYSIS_FORECAST_PHY_001_024/INFORMATION)) to the `catalog.xml` with:

```
    <datasetRoot path="datasets" location="/usr/local/tds/datasets" />

    <dataset name="cmems" ID="cmems"
             serviceName="all"  urlPath="datasets/cmems/cmems_forecast_20210101.nc" dataType="Grid"/>
```
The new dataset  was recognized just fine by the OpenDAP, HTTPServer and some other services, but there were some problems with the NCSS <s>and WMS services.</s>

# NCSS

The geographical region shown in the map selection is not the same as the region in the grid. I tried some other files and the region extent never changed.
![region](https://user-images.githubusercontent.com/13545615/137540928-b8e254c7-13dc-4e93-8c4e-6dbf5f09df0b.png)

When using the "NetCDF Subset Service for Grids As Points":
* [querying](http://localhost:8080/thredds/ncss/grid/datasets/cmems/cmems_forecast_20210101.nc?var=zos&latitude=-23&longitude=-45&time_start=2021-01-01T12:00:00Z&time_end=2021-01-03T12:00:00Z&timeStride=1&vertCoord=&accept=csv) only the 3D (time,lat,lon) variable "Sea surface height" the result was ok;
* [querying](http://localhost:8080/thredds/ncss/grid/datasets/cmems/cmems_forecast_20210101.nc?var=thetao&var=zos&latitude=-23&longitude=-45&time_start=2021-01-01T12%3A00%3A00Z&time_end=2021-01-03T12%3A00%3A00Z&timeStride=1&vertCoord=&accept=csv) a 3D and a 4D variable, only the first level of the 4D variable was returned. Previous (v4.?) versions of thredds returned all vertical levels of the 4D variable in this case.
* [querying](http://localhost:8080/thredds/ncss/grid/datasets/cmems/cmems_forecast_20210101.nc?var=thetao&latitude=-23&longitude=-45&time_start=2021-01-01T12%3A00%3A00Z&time_end=2021-01-03T12%3A00%3A00Z&timeStride=1&vertCoord=&accept=csv) only a 4D variable failed with error `IllegalArgumentException: Index out of range=3`
* querying a 2D variable (I did this with the `crossSeamProjection.nc` example file that comes with TDS) failed due to a forms time request ("Please fill out this field"), despite the file having no time dimesion.
![var2d](https://user-images.githubusercontent.com/13545615/137545868-add9c4d1-f1d9-4ee4-96a5-a0b4addb62be.png)

# <s>WMS</s> (solved, see post bellow)

Sometime the WMS fails with the error:
![wms2](https://user-images.githubusercontent.com/13545615/137550812-9782f5c6-527b-4138-b845-953e5c24748d.png)

while other times it seems to give a valid answer returning the Capabilities document:
![wms](https://user-images.githubusercontent.com/13545615/137546334-8c4a9534-ddbb-4039-aeb3-c9fc41ec8bc2.png)

but even in this case the ncWMS/Godiva3 viewer fails to load the data
![godiva](https://user-images.githubusercontent.com/13545615/137546553-5daedbba-cb16-47b2-a706-a034afcb17ef.png)
Note: I changed the Base Layer from "NaturalEarth WMS" to "NASA Blue Marble WMS" because the first seems to be offline for the last few days.

I also tried to use the getCapabilites response in QGIS but it also failed to be recognized:
![qgis](https://user-images.githubusercontent.com/13545615/137547118-34ef7774-7ea1-473f-8581-bc9532a4e1e8.png)

Thank you.

**Comment by marceloandrioni** [comment]
Hello, I just saw this [post](https://www.unidata.ucar.edu/blogs/news/entry/tds-users-update-your-configuration) in the main Unidata site and after replacing

`<!DOCTYPE wmsConfig SYSTEM "http://www.unidata.ucar.edu/schemas/thredds/dtd/ncwms/wmsConfig.dtd">`

with

`<!DOCTYPE wmsConfig SYSTEM "https://schemas.unidata.ucar.edu/thredds/dtd/ncwms/wmsConfig.dtd">`

the WMS (and Godiva3) worked just fine:

![tds_wms](https://user-images.githubusercontent.com/13545615/140557912-954202e5-4c40-40ef-b1f8-f16e17d23e6a.png)

The problem with the NCSS query (specifically the NetCDF Subset Service for Grids As Points) remains. Any idea of what may be the problem? 

Thank you.

**Comment by sharon-tickell** [comment]
Thank you for  noting that link about the WMS DTD, @marceloandrioni !  I just ran into that issue, and my google-fu had not turned up that post.   Your comment just saved my afternoon.

**Comment by marceloandrioni** [comment]
Hello @sharon-tickell, I am glad it was useful. I also lost a few hours until I stumbled in the Unidata post. If you happen to find a solution for the NCSS problem let know. Thank you.

**Comment by haileyajohnson** [comment]
Hi @marceloandrioni and @manoch-iris - I'm working on a fix for the grid as points issue, but for now I can give explanation of what's going on: in the current version, the TDS assumes that grid as point is requesting a timeseries of observations at a given point, which causes errors when there isn't a time dimension or when there are multiple observations for a given time (e.g. a vertical dimension).

This will be fixed, along with the map widget, before we release 5.4.

**Comment by manoch-iris** [comment]
Thank you, Hailey! This fix will help us a lot since our data does not have a time dimension.

Thanks again,

—manoch

**Comment by marceloandrioni** [comment]
Hello @haileyajohnson, I am glad to hear the issue will be fixed in the next release. In case you need some test file, there is a link in my first post to one from Copernicus Marine (with 3 times, 3 depths and covering part of the western South Atlantic).

Thank you.

**Comment by haileyajohnson** [comment]
This should be fixed now in the latest TDS snapshot release, and will be included in the upcoming 5.4 release (ETA any day now). Grid as point fixed in [netcdf-java #986](https://github.com/Unidata/netcdf-java/pull/986) and #218 (as well a few other related PRs)

map widget fixed in #221 

Thank you both for helping and providing data to test on!
